src/components/DataCollector/index.js

```diff
@@ -68,6 +74,7 @@ const createDataCollector = ({ config, logger, network }) => {
         // it overlays on top of any data Alloy automatically
         // provides. This allows the user to override the
         // automatically collected data.
+        event.mergeMeta(meta);
```

Comment:
@jonsnyder I am not against having a different mechanism for appending `meta` or any other data to the outgoing request. However this code used to  work and probably when `event` command was switched from ordinary `data` to `xdm` and `data`, this functionality has been broken.

Reply:
@jfkhoury and @jonsnyder if we don't want to expose `meta`, it will still be part of the public API exposed by Konductor and we will have to document that anyway.

Reply:
I'd rather not expose `meta` to the user until we understand the use case. I understand it will be a part of Konductor's API and will need to be documented there, but that doesn't mean it needs to be a part of Alloy's and I'm all about reducing the cognitive load for the Alloy user where possible.

I think this is pretty easily solvable by changing:

```
commands: {
   event: createEventHandler
}
```

to something like:

```
commands: {
 event(options) {
    const { viewStart, documentUnloading, xdm, data }) = options;
    // We don't want to allow customers to pass a meta option.
    return createEventHandler({
      viewStart,
      documentUnloading,
      xdm,
      data
    });
  }
}
```

I would also change `createEventHandler` to not pass `options` into `lifecycle.onBeforeEvent`. I don't think it provides value and we don't seem to be using it anywhere.

Another option, which is part of a larger validation effort that we should do anyway, is we should be validating options for all commands, similar to how we're validating options passed into the `configure` command. If we did that, and we failed validation on extraneous options, then we could set up validation for the `event` command to not accept a `meta` option.

Reply:
Leaving as is for now. Hopefully will come with a better approach in the next commit.